### DISCUSSION TITLE
Authentication Issues When Accessing REST API.

### DISCUSSION THREAD

**Discussion by DarkTinia** [keycloak/keycloak#25469] [Q&A] [answered]
I obtained the token using the following method.
![1](https://github.com/keycloak/keycloak/assets/51089199/f4ed58f7-da03-445c-aadd-257bba9342f9)
JWT.io show that.
![QQ截图20231211200145](https://github.com/keycloak/keycloak/assets/51089199/ae309bde-ae11-435a-8b55-699a3bdd0181)
But when I use this token to access the API, I tried two methods, and both of them returned a 401 error.
![3](https://github.com/keycloak/keycloak/assets/51089199/de62e13e-7edb-4709-ba6e-cf204161152e)
![5](https://github.com/keycloak/keycloak/assets/51089199/7ac882f6-8cfd-46d6-85ab-3a85bea5d065)
I followed the steps in a YouTube video carefully, without skipping any of the instructions. I'm not sure why I keep getting a 401 error when trying to access the API.  Could you please help me, I really appreciate any assistance.

**Comment by martinleim** [comment]
Have a look at the Keycloak log, it prints some information why the authentication failed.

**Comment by danielFesenmeyer** [comment]
It seems like the issuer (`iss` claim) does not match the URL used for the request. You either should use the same URL or configure _hostname_ appropriately: https://www.keycloak.org/server/hostname

**Reply by danielFesenmeyer** [reply, marked as answer]
The problem seems to be the different port.

**Reply by DarkTinia** [reply]
I'm sorry for the delayed response, and thank you very much! The issue has been resolved, and I no longer encounter this problem when accessing through the hostname (localhost).
